GenericJSONTests/QueryingTests.swift

```diff
@@ -58,4 +58,27 @@ class QueryingTests: XCTestCase {
         XCTAssertEqual(json.foo, JSON.string("bar"))
         XCTAssertEqual(json.nonesuch, nil)
     }
+    
+    func testKeyPath() {
+        let json: JSON = [
+            "string": "foo bar",
+            "boolean": true,
+            "number": 123,
+            "object": [
+                "str": "col",
+                "arr": [1, 2, 3],
+                "obj": [
+                    "x": "rah",
+                    "y": "tar",
+                    "z": "yaz"
+                ]
+            ]
+        ]
+        XCTAssertEqual(json[keyPath: "string"], JSON.string("foo bar"))
```

Comment:
This can be simplified to:

```swift
XCTAssertEqual(json[keyPath: "string"], "foo bar")
XCTAssertEqual(json[keyPath: "boolean"], true)
XCTAssertEqual(json[keyPath: "number"], 123)
XCTAssertEqual(json[keyPath: "object.str"], "col")
XCTAssertEqual(json[keyPath: "object.arr"], [1, 2, 3])
XCTAssertEqual(json[keyPath: "object.obj.y"], "tar")
```

This is enabled by the `ExpressibleByXXXLiteral` that makes it possible for the compiler to “automatically upgrade” a plain value to `JSON` when an instance of `JSON` is expected. Here, the compiler can see that the second argument to `XCTAssertEqual` has to be of the same type as the first one, and since the first one is a `JSON?` (as we are calling `JSON.subscript(keyPath:)`), the second is automatically converted to `JSON`, too.

In other words, conforming `JSON` to the `ExpressibleByXXXLiteral` protocols makes it possible for plain value literals such as `"foo"`, `true` or `42` to express JSON values without any extra fluff. Which kind of makes sense and is nice.

Reply:
Thanks, I wasn't aware of the automatic upgrade the compiler does. I have updated the tests.